compiler/frontend/beam/src/main/java/org/apache/nemo/compiler/frontend/beam/NemoPipelineResult.java

```diff
@@ -49,10 +60,12 @@ public State cancel() throws IOException {
 
   @Override
   public State waitUntilFinish(final Duration duration) {
-    throw new UnsupportedOperationException();
-    // TODO #208: NemoPipelineResult#waitUntilFinish hangs
-    // Previous code that hangs the job:
-    // return (State) super.waitUntilJobFinish(duration.getMillis(), TimeUnit.MILLISECONDS);
+    try {
+      this.jobDone.await();
+    } catch (final InterruptedException e) {
+      throw new RuntimeException(e);
+    }
+    return (State) super.waitUntilJobFinish(duration.getMillis(), TimeUnit.MILLISECONDS);
```

Comment:
@johnyangk I'm not sure if this line actually does anything. As the original author of the code here, what do you think?

Reply:
Does it actually get called after the job finishes? I was unsure, because I experienced failing integration tests when using this by itself.

Reply:
Yes, there seems to be a bug. I also experienced the same issue.

JIRA:
https://issues.apache.org/jira/browse/NEMO-208

Reply:
Does this function call actually shut down the nemo driver and executor?